examples/add_reverb_to_file.py

```diff
@@ -0,0 +1,135 @@
+        ),
+        default=None,
+    )
+
+    # Instantiate the Reverb object early so we can read its defaults for the argparser --help:
+    reverb = Reverb()
+
+    parser.add_argument("--room-size", type=float, default=reverb.room_size)
+    parser.add_argument("--damping", type=float, default=reverb.damping)
+    parser.add_argument("--wet-level", type=float, default=reverb.wet_level)
+    parser.add_argument("--dry-level", type=float, default=reverb.dry_level)
+    parser.add_argument("--width", type=float, default=reverb.width)
+    parser.add_argument("--freeze-mode", type=float, default=reverb.freeze_mode)
+
+    parser.add_argument(
+        "--cut-reverb-tail",
+        action="store_true",
+        help=(
+            "If passed, remove the reverb tail to the end of the file. "
+            "The output file will be identical in length to the input file."
+        ),
+    )
+    args = parser.parse_args()
+
+    for arg in ('room_size', 'damping', 'wet_level', 'dry_level', 'width', 'freeze_mode'):
+        setattr(reverb, arg, getattr(args, arg))
+
+    if not args.output_file:
+        args.output_file = args.input_file + ".reverb.wav"
+
+    sys.stderr.write(f"Opening {args.input_file}...\n")
+
+    with sf.SoundFile(args.input_file) as input_file:
+        sys.stderr.write(f"Writing to {args.output_file}...\n")
+        with sf.SoundFile(
+            args.output_file,
+            'w',
+            samplerate=input_file.samplerate,
+            channels=input_file.channels,
+        ) as output_file:
+            length = get_num_frames(input_file)
+            length_seconds = length / input_file.samplerate
+            sys.stderr.write(f"Adding reverb to {length_seconds:.2f} seconds of audio...\n")
+            with tqdm(
+                total=length_seconds,
+                desc="Adding reverb...",
+                bar_format=(
+                    "{percentage:.0f}%|{bar}| {n:.2f}/{total:.2f} seconds processed"
+                    " [{elapsed}<{remaining}, {rate:.2f}x]"
+                ),
+                # Avoid a formatting error that occurs if
+                # TQDM tries to print before we've processed a block
+                delay=1000,
+            ) as t:
+                for dry_chunk in input_file.blocks(BUFFER_SIZE_SAMPLES, frames=length):
+                    # Actually call Pedalboard here:
+                    # (reset=False is necessary to allow the reverb tail to
+                    # continue from one chunk to the next.)
+                    effected_chunk = reverb.process(
+                        dry_chunk, sample_rate=input_file.samplerate, reset=False
+                    )
+                    # print(effected_chunk.shape, np.amax(np.abs(effected_chunk)))
+                    output_file.write(effected_chunk)
+                    t.update(len(dry_chunk) / input_file.samplerate)
+                    t.refresh()
+            if not args.cut_reverb_tail:
+                while True:
```

Comment:
An alternative condition idea - pull until you match the number of bytes input. A little fear I have with the break condition you set is that what if a user intentionally inputs silence?